From 0088fe022c3ec4d35d3f0b098b48bdc2eadd5cb7 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Fri, 12 Sep 2014 18:09:41 -0400
Subject: [virt] kvm/ppc: book3s hv - Return ENODEV error rather than EIO

Message-id: <1410545655-205645-353-git-send-email-dzickus@redhat.com>
Patchwork-id: 94440
O-Subject: [RHEL7 PATCH 352/626] KVM: PPC: Book3S HV: Return ENODEV error rather than EIO
Bugzilla: 1127366
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1127366

commit 739e2425fea6349ac674e93648953b3a08985f2f
Author: Paul Mackerras <paulus@samba.org>
Date:   Tue Mar 25 10:47:05 2014 +1100

    KVM: PPC: Book3S HV: Return ENODEV error rather than EIO

    If an attempt is made to load the kvm-hv module on a machine which
    doesn't have hypervisor mode available, return an ENODEV error,
    which is the conventional thing to return to indicate that this
    module is not applicable to the hardware of the current machine,
    rather than EIO, which causes a warning to be printed.

    Signed-off-by: Paul Mackerras <paulus@samba.org>
    Acked-by: Scott Wood <scottwood@freescale.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/kvm/book3s_hv.c b/arch/powerpc/kvm/book3s_hv.c
index 5fd1c13..4bace7d 100644
--- a/arch/powerpc/kvm/book3s_hv.c
+++ b/arch/powerpc/kvm/book3s_hv.c
@@ -2459,7 +2459,7 @@ static int kvmppc_book3s_init_hv(void)
   */
  r = kvmppc_core_check_processor_compat_hv();
  if (r < 0)
-  return r;
+  return -ENODEV;
 
  kvm_ops_hv.owner = THIS_MODULE;
  kvmppc_hv_ops = &kvm_ops_hv;
-- 
1.7.1